core/scripts.py

```diff
@@ -2668,6 +2668,96 @@ def process_loan(self, loan: Loan):
             self.notifications.send_loan_expiry_message(loan, delta.days, tokens)
 
 
+class SuppressWorkForLibraryScript(Script):
+    """Suppress works from a library by identifier"""
+
+    BY_DATABASE_ID = "Database ID"
+
+    @classmethod
+    def arg_parser(cls, _db: Session | None) -> argparse.ArgumentParser:  # type: ignore[override]
+        parser = argparse.ArgumentParser()
+        if _db is None:
+            raise ValueError("No database session provided.")
+        library_name_list = sorted(str(l.short_name) for l in _db.query(Library))
+        library_names = '"' + '", "'.join(library_name_list) + '"'
+        parser.add_argument(
+            "-l",
+            "--library",
+            help="Short name of the library. Libraries on this system: %s."
+            % library_names,
+            required=True,
+            metavar="SHORT_NAME",
+        )
+        parser.add_argument(
+            "-t",
+            "--identifier-type",
+            help="Identifier type (default: ISBN). "
+            f'To name identifiers by their database ID, use --identifier-type="{cls.BY_DATABASE_ID}".',
+            default="ISBN",
+        )
+        parser.add_argument(
+            "-i",
+            "--identifier",
+            help="The identifier to suppress.",
+            required=True,
+        )
```

Comment:
Minor: Since both library short name and identifier are both required, could we use the following syntax?

```
command [-i id-type] library identifiers...
```

That would mean a some reworking below and in the tests, but might make our work a little easier when we have to suppress more than one title with same `id-type`.

Reply:
I'd like to leave this to a follow up. I did the easiest thing here, and just re-implemented some bits of the `IdentifierScript` and `LibraryScript` types, since this script needs both. I think a better solution would be to update those base classes to be able to cooperate or just be mixin classes, but I was feeling like I had already brought in too much tangential refactoring already for this ticket, so I just left it.

I'll make a follow up ticket for that, and once thats done this can just be an IdentiferScript that can take multiple identifiers.